### ISSUE TITLE
Is it possible to remove the confirmation of receipt of the package by the client?

### ISSUE THREAD

**Issue by whakru** [Links2004/arduinoWebSockets#536]
I have difficulties when the client starts to think longer then webSocket.loop () starts to be very large in time. And that breaks my code after some time.
Inclusion of Async is impossible for me, it breaks me SoftwareSerial.
Is it possible to disable the package confirmation by the client? and where to fix it.
or at least put a restriction on the execution of the webSocket.loop () function for a certain time and no more.

**Comment by Links2004** [comment]
same problem as #535?
package confirmation is part of the TCP stack, not much we can do here.
if you have critical tasks you can stop calling the loop for some time, but keep in mind that most 
websocket server / clients send a ping pong and close the connection when they get no answer.
on top of that the TCP layer buffer can get full when the loop is not called for a long time.

**Comment by whakru** [comment]
> on top of that the TCP layer buffer can get full when the loop is not called for a long time.
yes this is also a problem.  But is it possible to clear the buffer manually?

**Comment by Links2004** [comment]
yes and no, you access the tcp socket you can remove the data, but this will break the websocket connection since, the protocol is designed in a way that you need all the information to see where one message ends and the other starts.

**Comment by whakru** [comment]
I do not think that delay (0);  can help here.  the problem is that the client sometimes takes a long time to respond.  and the web socket cycle has been waiting and overflowing all this time.  Is there any way to limit this waiting time from the client?  let's set the time for an answer yourself?  so that the connection is terminated if the client does not respond, let’s say more than some time.  and still interested in cleaning the buffer manually.

**Comment by Links2004** [comment]
not sure but you looking for the TCP timeout?
https://github.com/Links2004/arduinoWebSockets/blob/master/src/WebSockets.h#L87

**Comment by whakru** [comment]
ok, i will try to set WEBSOCKETS_TCP_TIMEOUT 100ms. will it disconnect all clients after time? or only those who thought about the answer for a long time?

**Comment by Links2004** [comment]
the timeout is per client

**Comment by whakru** [comment]
decreasing WEBSOCKETS_TCP_TIMEOUT did not work.  when I set 150ms.  the client could not connect.  it was instantly dumped.  I would like to change not the general timeout, but the timeout of sending the data packet.

**Comment by Links2004** [comment]
the lib uses the setTimeout of the TCP socket and sets it with the WEBSOCKETS_TCP_TIMEOUT define.
if this is not working for you, then the problem is most likely in the TCP stack and not the websocket library.
does the ESP and or library debug output show anything?

**Comment by whakru** [comment]
My code is designed so that the free time is about 100-200ms and I would not want the sending to take more time.  in my case it is not possible to increase the wait time.  I am doing a communication project with a different protocol that uses a soft serial port.  and there are used time-limited functions.  And if the client of the web socket responds to the sent message for a long time> 100-200ms, then my code becomes useless.  and I’d like to limit in time exactly the response to the sent packet.  so that the esp does not wait for a confirmation response for too long.  (or maybe somehow ignored him. something like sent and forgot).  asynchronous transfer library breaks software serial port.  the developer is not responding.

**Comment by whakru** [comment]
is there any way to remove from the sending webSocket.sendTXT waiting for a response from the client?

**Comment by Links2004** [comment]
the `WifiClient` API does not provide a feature like this as fare I know.
all TCP writes of the lib are ending up in the function here:
https://github.com/Links2004/arduinoWebSockets/blob/master/src/WebSockets.cpp#L656
which ultimately call write of the WiFiClient
https://github.com/Links2004/arduinoWebSockets/blob/master/src/WebSockets.cpp#L681


https://github.com/esp8266/Arduino/blob/master/libraries/ESP8266WiFi/src/WiFiClient.h#L59-L61
